core/src/main/java/com/linecorp/armeria/client/limit/ConcurrencyLimit.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.limit;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Predicate;
+
+import com.google.common.base.MoreObjects;
+
+import com.linecorp.armeria.client.ClientRequestContext;
+
+/**
+ * Concurrency settings that limits the concurrent number of active requests.
+ */
+public final class ConcurrencyLimit {
+    /**
```

Comment:
How about also adding
```java
public static ConcurrencyLimit of() {
    return builder().build();
}
```
that builds `ConcurrencyLimit` with the default properties?
By the way, do you have sensible default for `maxConcurrency`?

Reply:
I think it is hard to tell what is a sensible default on client-side.
Because the `maxConcurrency` depends on the capacity of a server.
How about making `maxConcurrency` a mandatory field?
```java
ConcurrencyLimit.of(maxConcurrency);
ConcurrencyLimit.builder(maxConcurrency)
                .timeoutMillis(...)
                ...
                .build();
```

Reply:
Yes, it makes sense to make it mandatory.

Reply:
We conventionally prefer `builder()` for mandatory fields over `build()`. 
https://github.com/line/armeria/blob/32eabd2eb1f1be73f03a6425fa193631064cf2db/core/src/main/java/com/linecorp/armeria/common/ContentDisposition.java#L79-L83